### ISSUE TITLE
Warning message from Dialog ( The prop `text` is marked as required in `Button` )

### ISSUE THREAD

**Issue by m-samkc** [cerner/terra-core#1437]
# Issue Description
Dialog component throws warning message for `text` prop required in Button. 

## Issue Type
- Bug

## Expected Behavior
Dialog component is using Button component for adding button features which has `text` as required prop. However while using Dialog component use doesn't have option to pass the prop down to Button.

## Current Behavior
```
Warning: Failed prop type: The prop `text` is marked as required in `Button`, but its value is `undefined`.
    in Button (created by Dialog)
    in Dialog (at EmbeddedModalContent.js:38)
    in EmbeddedModalContent (at EmbeddedModalManager.js:37)
    in div (created by Slide)
    in Slide (created by SlideGroup)
    in Transition (created by CSSTransition)
    in CSSTransition (created by SlideGroup)
    in div (created by TransitionGroup)
    in TransitionGroup (created by SlideGroup)
    in SlideGroup (created by DisclosureManager)
    in div (created by ModalContent)
    in div (created by FocusTrap)
```

## Steps to Reproduce
1. Use Dialog component 
2. Check chrome debugger for warning message

**Comment by emilyrohrbough** [comment]
Duplicate of https://github.com/cerner/terra-core/issues/1351

**Comment by bjankord** [comment]
Resolved in https://github.com/cerner/terra-core/pull/1455
